crates/bevy_scene/src/dynamic_scene.rs

```diff
@@ -99,7 +99,7 @@ impl DynamicScene {
             // Apply/ add each component to the given entity.
             for component in &scene_entity.components {
                 let registration = type_registry
-                    .get_with_name(component.type_name())
+                    .get_with_name(component.type_name().as_ref())
```

Comment:
I don't think we need this `as_ref` here either (there's also a number of other ones I missed that could probably be removed as well).